src/main/java/com/gmail/chernii/oleksii/dao/CompanyDAO.java

```diff
@@ -0,0 +1,63 @@
+package com.gmail.chernii.oleksii.dao;
+
+import com.gmail.chernii.oleksii.entity.Company;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+
+/**
+ * Created by Space on 16.04.2019.
+ */
+@Log4j
+@AllArgsConstructor
+public class CompanyDAO implements DAO<Company> {
```

Comment:
Why DAO, not Dao?
https://mate-academy.github.io/style-guides/java/java.html#s5.2.2-class-names